### DISCUSSION TITLE
Issue with Internet Access via RaspAP: wlan1 connected, but no access through wlan0

### DISCUSSION THREAD

**Discussion by fgordon** [RaspAP/raspap-webgui#1706] [Q&A] [answered]
Hello everyone,

I’m using a Raspberry Pi 4 Model B Rev 1.4 with RaspAP to set up a mobile WiFi access point. Here’s my setup:

**wlan0:** The Raspberry Pi’s internal WiFi module, configured as a hotspot.
**wlan1:** An external USB WiFi adapter (Ralink RT5572), successfully connected to a WiFi network with Internet access.
For installation, I used the latest Raspberry Pi OS (bookworm arm64 lite) and followed the instructions in the documentation to install RaspAP via the 'Quick Installer.'

In the RaspAP web interface, wlan1 is shown as having Internet access. However, wlan0 does not. Devices connecting to the RaspAP hotspot (wlan0) also fail to access the Internet, showing an error that no Internet connection is available.

I’ve attached the current routing table and network configuration below.

**Routing Table:**
```
default via 10.3.141.1 dev wlan0 src 10.3.141.1 metric 3003
default via 192.168.179.1 dev wlan1 proto dhcp src 192.168.179.3 metric 3006
10.3.141.0/24 dev wlan0 proto dhcp scope link src 10.3.141.1 metric 3003
192.168.179.0/24 dev wlan1 proto dhcp scope link src 192.168.179.3 metric 3006
```

**Network Configuration:**
_eth0_
```
2: eth0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc mq state DOWN group default qlen 1000
    link/ether dc:a6:32:f6:c9:2b brd ff:ff:ff:ff:ff:ff
```
_wlan0_
```
3: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq state UP group default qlen 1000
    link/ether dc:a6:32:f6:c9:2c brd ff:ff:ff:ff:ff:ff
    inet 10.3.141.1/24 brd 10.3.141.255 scope global noprefixroute wlan0
       valid_lft forever preferred_lft forever
    inet6 fe80::60e0:6eaa:7824:1619/64 scope link
       valid_lft forever preferred_lft forever
```

_wlan1_
```
6: wlan1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether dc:4e:f4:08:6a:a3 brd ff:ff:ff:ff:ff:ff
    inet 192.168.179.3/24 brd 192.168.179.255 scope global dynamic noprefixroute wlan1
       valid_lft 19890sec preferred_lft 17190sec
    inet6 2a02:3100:1ae7:1801:e630:5a1d:c21c:6d79/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 7043sec preferred_lft 3443sec
    inet6 fe80::2453:eea3:ae50:1a09/64 scope link
       valid_lft forever preferred_lft forever
```

Does anyone have an idea what might be causing this? Could it be a missing routing or IP forwarding configuration? I’d really appreciate your help!

Thanks in advance!

**Comment by feh123** [comment]
Hi I have had a similar problem with a wlan0/wlan1 setup on a RPI4 that I use to avoid needing a wired connection. I found that Bookworm always failed to give me even an access point. I fixed this by going back to Bullseye Lite 32 or 64Bit. I was using a different second wifi from https://thepihut.com/products/dual-band-usb-wifi-adapter-for-raspberry-pi-5ghz-2-4ghz-mt7612u with a MT7612U chipset
I have found that the RPI 5 with wlan0/wlan1 will work on Bookworm with this adapter.

**Comment by billz** [comment]
Refer to [configuring routing](https://docs.raspap.com/repeater/#configure-routing) for a WiFi repeater.

There is no limitation to using Bookworm. This project has been validated on both [Bookworm and Bullseye](https://docs.raspap.com/#compatible-operating-systems).  I'm connected to an AP on Bookworm 64-bit Desktop (testing only, Lite is recommended). No issues whatsoever. Re: adapters, the [usual FAQ](https://docs.raspap.com/faq/#adapters) applies.

**Reply by fgordon** [reply, marked as answer]
> Refer to [configuring routing](https://docs.raspap.com/repeater/#configure-routing) for a WiFi repeater.
> 
> There is no limitation to using Bookworm. This project has been validated on both [Bookworm and Bullseye](https://docs.raspap.com/#compatible-operating-systems). I'm connected to an AP on Bookworm 64-bit Desktop (testing only, Lite is recommended). No issues whatsoever. Re: adapters, the [usual FAQ](https://docs.raspap.com/faq/#adapters) applies.

Thank you for pointing me in the right direction regarding the routing configuration for a WiFi repeater!

Devices connected to the RaspAP hotspot (wlan0) can access the Internet via the external adapter (wlan1) without any problems.

I appreciate the clarification about Bookworm compatibility. Everything is now working perfectly.

Thanks again for the support and guidance!

**Reply by billz** [reply]
@fgordon glad to hear it!

RaspAP doesn't know that you're creating a repeater unless you explicitly [enable this mode](https://docs.raspap.com/repeater/#alternate-routing-method). However, the manual method using interface metrics isn't terribly difficult. It also has the added benefit of giving some insight into Linux networking 😉
